tools/proofers/preview.js

```diff
@@ -895,12 +956,42 @@ $(function () {
             if (viewMode !== "show_tags") {
                 colorString = colorString0 + '>$1</span>';
             }
-            // sub- and super-scripts
-            txt = txt.replace(/_\{([^\}]+)\}/g, '<span class="sub"' + colorString);
-            txt = txt.replace(/\^\{([^\}]+)\}/g, '<span class="sup"' + colorString);
-            // single char superscript -  any char except {
-            // do not allow < incase it's a tag which would screw up
-            txt = txt.replace(/\^([^\{<])/g, '<span class="sup"' + colorString);
+
+            // process sub- and super-scripts
+            function ssProcess(text) {
+                text = text.replace(/_\{(.+?)\}/g, '<span class="sub"' + colorString);
+                text = text.replace(/\^\{(.+?)\}/g, '<span class="sup"' + colorString);
+                // single char superscript -  any char except {
+                // do not allow < incase it's a tag which would screw up
+                return text.replace(/\^([^{<])/g, '<span class="sup"' + colorString);
```

Comment:
this function can't decide if it is immutable or not it seems. If we return the text, why modify the argument? I'd prefer to not modify arguments in general.

Reply:
This was written several months ago. Itdoes seem a bit strange. I'll look at it again.